File: AccuPay/Domain/Payroll/AllowancePolicy.vb
Option Strict On

Public Class AllowancePolicy

    Private ReadOnly _settings As ListOfValueCollection

    Public Sub New(settings As ListOfValueCollection)
        _settings = settings
    End Sub

    Public ReadOnly Property IsLeavePaid As Boolean
        Get
            Return _settings.GetBoolean("AllowancePolicy.IsLeavePaid")
        End Get
    End Property

    Public ReadOnly Property IsOvertimePaid As Boolean
        Get
            Return _settings.GetBoolean("AllowancePolicy.IsOvertimePaid")
        End Get
    End Property

    Public ReadOnly Property IsRestDayPaid As Boolean
        Get
            Return _settings.GetBoolean("AllowancePolicy.IsRestDayPaid")
        End Get
    End Property

    Public ReadOnly Property IsRestDayOTPaid As Boolean
        Get
            Return _settings.GetBoolean("AllowancePolicy.IsRestDayOTPaid")
        End Get
    End Property

    Public ReadOnly Property IsSpecialHolidayPaid As Boolean
        Get
            Return _settings.GetBoolean("AllowancePolicy.IsSpecialHolidayPaid")
        End Get
    End Property

    Public ReadOnly Property IsRegularHolidayPaid As Boolean
        Get
            Return _settings.GetBoolean("AllowancePolicy.IsRegularHolidayPaid")
        End Get
    End Property

End Class

File: AccuPay.Core/Services/Policies/AllowancePolicy.cs
namespace AccuPay.Core.Services
{
    public class AllowancePolicy
    {
        private readonly ListOfValueCollection _settings;

        public AllowancePolicy(ListOfValueCollection settings) => _settings = settings;

        public bool IsLeavePaid => _settings.GetBoolean("AllowancePolicy.IsLeavePaid");

        public bool IsOvertimePaid => _settings.GetBoolean("AllowancePolicy.IsOvertimePaid");

        public bool IsRestDayPaid => _settings.GetBoolean("AllowancePolicy.IsRestDayPaid");

        public bool IsRestDayOTPaid => _settings.GetBoolean("AllowancePolicy.IsRestDayOTPaid");

        public bool IsSpecialHolidayPaid => _settings.GetBoolean("AllowancePolicy.IsSpecialHolidayPaid");

        public bool IsRegularHolidayPaid => _settings.GetBoolean("AllowancePolicy.IsRegularHolidayPaid");

        public bool HolidayAllowanceForMonthly => _settings.GetBoolean("AllowancePolicy.HolidayAllowanceForMonthly");

        public bool NoPremium => _settings.GetBoolean("AllowancePolicy.NoPremium");

        public string CalculationType => _settings.GetString("AllowancePolicy.CalculationType");
    }
}
